test/e2eprovider/server/server_test.go

```diff
@@ -0,0 +1,138 @@
+// +build e2e
+
+package server
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"reflect"
+	"testing"
+
+	"sigs.k8s.io/secrets-store-csi-driver/provider/v1alpha1"
+)
+
+var (
+	testMockServer *Server
+)
+
+func TestMain(m *testing.M) {
+	setup()
+	exitCode := m.Run()
+	os.Exit(exitCode)
+}
+
+func setup() {
+	tempDir, err := ioutil.TempDir("", "")
+	if err != nil {
+		panic(err)
```

Comment:
pass `t *testing.T` as arg to the function and use `t.Error` instead

Reply:
We do not have access to `testing.T` in `TestMain` method